tests/components/apcupsd/test_config_flow.py

```diff
@@ -153,3 +155,126 @@ async def test_flow_minimal_status(
         assert result["data"] == CONF_DATA
         assert result["title"] == expected_title
         mock_setup.assert_called_once()
+
+
+async def test_reconfigure_flow_works(hass: HomeAssistant) -> None:
+    """Test successful reconfiguration of an existing entry."""
+    mock_entry = MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="APCUPSd",
+        data=CONF_DATA,
+        unique_id=MOCK_STATUS["SERIALNO"],
+        source=SOURCE_USER,
+    )
+    mock_entry.add_to_hass(hass)
+
+    # Start the reconfigure flow.
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_RECONFIGURE, "entry_id": mock_entry.entry_id},
+    )
```

Comment:
Instead you can do `await entry.start_reconfigure_flow(hass)`

Reply:
Didn't know that, updated all test cases here. Thx!